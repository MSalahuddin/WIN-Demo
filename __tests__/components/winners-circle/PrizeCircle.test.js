import React from 'react';
import { render } from 'react-native-testing-library';

import PrizeCircle from '../../../js/components/winners-circle/PrizeCircle';

describe('Component PrizeCircle', () => {
  test('renders correctly when type is swap', () => {
    const output = render(<PrizeCircle imageUrl="a" type="SWAP" />);
    expect(output).toMatchSnapshot();
  });

  test('renders correctly', () => {
    const output = render(<PrizeCircle imageUrl="a" />);
    expect(output).toMatchSnapshot();
  });
});
