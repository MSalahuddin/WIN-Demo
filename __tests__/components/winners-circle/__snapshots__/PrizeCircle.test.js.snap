// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PrizeCircle renders correctly 1`] = `
<View>
  <View
    circleColor="#5EC98C"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C4D1FA",
          "borderColor": "#5EC98C",
          "borderRadius": 85,
          "borderWidth": 2,
          "height": 170,
          "justifyContent": "center",
          "width": 170,
        },
      ]
    }
  >
    <Image
      fullCircle={false}
      resizeMode="cover"
      source={
        Object {
          "uri": "a",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 82,
            "height": 164,
            "width": 164,
          },
        ]
      }
    />
  </View>
  <View
    circleColor="#5EC98C"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#5EC98C",
          "borderColor": "#5EC98C",
          "borderRadius": 24,
          "borderWidth": 2,
          "bottom": -22,
          "height": 44,
          "justifyContent": "center",
          "left": 44,
          "position": "absolute",
          "width": 88,
        },
      ]
    }
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#FFFFFF"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#FFFFFF",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 8,
          },
        ]
      }
    >
      x1
    </Text>
  </View>
</View>
`;

exports[`Component PrizeCircle renders correctly when type is swap 1`] = `
<View>
  <View
    circleColor="#5EC98C"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C4D1FA",
          "borderColor": "#5EC98C",
          "borderRadius": 85,
          "borderWidth": 2,
          "height": 170,
          "justifyContent": "center",
          "width": 170,
        },
      ]
    }
  >
    <Image
      fullCircle={true}
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "borderRadius": 0,
            "height": 64,
            "width": 98,
          },
        ]
      }
    />
  </View>
  <View
    circleColor="#5EC98C"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#5EC98C",
          "borderColor": "#5EC98C",
          "borderRadius": 24,
          "borderWidth": 2,
          "bottom": -22,
          "height": 44,
          "justifyContent": "center",
          "left": 44,
          "position": "absolute",
          "width": 88,
        },
      ]
    }
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#FFFFFF"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#FFFFFF",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 8,
          },
        ]
      }
    >
      x1
    </Text>
  </View>
</View>
`;
